File: backend/shared/uidl/src/main/java/io/mateu/uidl/annotations/Text.java
package io.mateu.uidl.annotations;

import io.mateu.uidl.data.TextContainer;
import io.mateu.uidl.data.TextSize;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/** Created by miguel on 18/1/17. */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.ANNOTATION_TYPE})
public @interface Text {

  TextContainer container() default TextContainer.p;

  /**
   * Font size of the rendered text. {@code m} (the default) applies nothing; the other values
   * enlarge ({@code l}, {@code xl}) or reduce ({@code s}, {@code xs}) the font.
   */
  TextSize size() default TextSize.m;

  /**
   * When {@code true} the rendered container drops its block margins ({@code margin-block-start}
   * and {@code margin-block-end} to 0) — e.g. so a {@code <p>} caption sits tight against its
   * neighbours. Independent of {@link #size()}.
   */
  boolean noMargins() default false;
}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Text.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import { TextContainer } from "@mateu/shared/apiClients/dtos/componentmetadata/TextContainer";

export default interface Text extends ComponentMetadata {

    container: TextContainer
    text: string
    /** Font size: xl | l | m | s | xs. m (or absent) applies nothing. */
    size?: string
    /** Drops the container's block margins (margin-block-start/end: 0). Independent of size. */
    noMargins?: boolean

}